[DSM] DDP-7556_mercury_sequencing

pegahtah opened the pull request:
Everything from branch DDP-7556_mercury_order is moved to this
---  update:
Tickets in this PR
https://broadinstitute.atlassian.net/browse/DDP-7556 
https://broadinstitute.atlassian.net/browse/DDP-7604
https://broadinstitute.atlassian.net/browse/DDP-7608
https://broadinstitute.atlassian.net/browse/DDP-8115
https://broadinstitute.atlassian.net/browse/DDP-8116

pegahtah committed:
DDP-7556 New Branch made to avoid formatting conflicts,
Everything from branch DDP-7556_mercury_order is moved to this

pegahtah committed:
adding `sequencing_restriction` and collection_date field

pegahtah committed:
adding condition check for sequencingRestrictions on Normal samples

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

pegahtah committed:
adding a check for sequencingRestriction

pegahtah committed:
DDP-7556 adding a check for having sequencing

pegahtah committed:
DDP-7556 adding new columns for the new fields

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

pegahtah committed:
DDP-7556 linting

pegahtah committed:
adding scss file

SimoneMaiwald commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-order.model.ts:
```diff
@@ -0,0 +1,17 @@
+
+  constructor( public shortId: string, public sample: string, public orderId: string,
+               public orderDate: string, public statusDetail: string, public sampleType: string ) {
+    this.shortId = shortId;
+    this.sample = sample;
+    this.orderId = orderId;
+    this.orderDate = orderDate;
+    this.statusDetail = statusDetail;
+    this.sampleType = sampleType;
```
I think this is not needed, @alextanlan yes?

SimoneMaiwald commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.html:
```diff
@@ -0,0 +1,33 @@
+<ng-container *ngIf="!loading">
```
do we have something that is displayed while it is loading?

SimoneMaiwald commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.ts:
```diff
@@ -0,0 +1,92 @@
+    fields.push( 'status' );
+    fields.push( 'orderId' );
+    const date = new Date();
+    console.log( map );
```
please remove log

SimoneMaiwald commented on ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html:
```diff
@@ -49,6 +49,10 @@
           <li *ngIf="hasRole().allowedToHandleSamples()"><a [routerLink]="['labelSettings']"> Label Settings </a></li>
+          <hr *ngIf="hasRole().allowedToHandleSamples()">
+          <li *ngIf="hasRole().allowedToHandleSamples() || hasRole().allowedToDoOrderSequencing()">
+            <a [routerLink]="['clinicalOrders']">Clinical Orders</a>
```
are we really calling that Clinical Orders?

pegahtah replied:
This is from the instructions in the doc Taylor has made

SimoneMaiwald commented on ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/models/sample.model.ts:
```diff
@@ -71,7 +75,7 @@ export class Sample {
       json.trackingNumberTo, json.trackingReturnId, json.kitLabel, testResults, json.upsTrackingStatus,
       json.upsReturnStatus, json.externalOrderStatus, json.externalOrderNumber, json.externalOrderDate,
-      json.careEvolve, json.uploadReason
+      json.careEvolve, json.uploadReason, json.collectionDate, json.sequencingRestrtiction, json.dsmKitRequestId
```
typo `sequencingRestrtiction`

pegahtah replied:
thanks!

SimoneMaiwald commented on ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts:
```diff
@@ -545,6 +549,38 @@ export class ParticipantPageComponent implements OnInit, OnDestroy, AfterViewChe
 
+  kitValueChanged( value: any, parameterName: string, sample: Sample ): void {
+    let v;
+    console.log(value);
```
remove

pegahtah committed:
DDP-7556 PR comments applied
some style changes

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

# Conflicts:
#	ddp-workspace/projects/ddp-dsm-ui/src/app/ALL-STUDIES/all-studies.module.ts
#	ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts
#	ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.html
#	ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts
#	ddp-workspace/projects/ddp-dsm-ui/src/app/services/role.service.ts

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.ts:
```diff
@@ -0,0 +1,91 @@
+      return 'Sent';
+    }
+    return statusDetail;
+
```
#Optional:
You could simply write here, like this:
`return  statusDetail  || 'Sent'`

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.ts:
```diff
@@ -0,0 +1,91 @@
+  public downloadList(): void {
+    const map: { shortId: string; sampleType: string; sample: string; orderDate: string; status: string; orderId: string }[] = [];
+    console.log( this.clinicalOrdersArray );
+    for (const order of this.clinicalOrdersArray) {
```
You can get rid of this console.log

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts:
```diff
@@ -1438,4 +1479,52 @@ export class ParticipantPageComponent implements OnInit, OnDestroy, AfterViewChe
+
+      },
+      error: () => {
+
```
#Optional
If we don't implement error handling, maybe to get rid of this as well

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts:
```diff
@@ -1438,4 +1479,52 @@ export class ParticipantPageComponent implements OnInit, OnDestroy, AfterViewChe
+    }
+    const enrolled: boolean = participant.data.status === 'ENROLLED';
+    let hasGender = false;
+    if (participant.oncHistoryDetails.find( onc => onc.gender !== '' && onc.gender !== undefined && onc.gender !== null )) {
```
#Optional
I would take this conditional check into other function and here I would just have a function call, like this:

```
if(isTrue()) {
}
```

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.ts:
```diff
@@ -0,0 +1,91 @@
+    const map: { shortId: string; sampleType: string; sample: string; orderDate: string; status: string; orderId: string }[] = [];
+    console.log( this.clinicalOrdersArray );
+    for (const order of this.clinicalOrdersArray) {
+      const dateCreated = this.getDateFormatted( order.orderDate );
```
#Optional:
You could also move this for loop's code block into a separate function and then call like this:
for (const order of this.clinicalOrdersArray) {
  doSomething(order)
}

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts:
```diff
@@ -1438,4 +1479,52 @@ export class ParticipantPageComponent implements OnInit, OnDestroy, AfterViewChe
+      const aboutYouActivity = participant.data.activities.find( activity => activity.activityCode === 'ABOUT_YOU' );
+      if (aboutYouActivity) {
+        const genderQuestion = aboutYouActivity.questionsAnswers.find( questionAnswer => questionAnswer.stableId === 'ASSIGNED_SEX' );
+        if (genderQuestion && genderQuestion.answer) {
```
I would store those strings in constants, or in enums.

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts:
```diff
@@ -0,0 +1,162 @@
+  currentPatchField: string;
+  errorMessage: string;
+  selectedNormal: SequencingOrder = new SequencingOrder( '', '', '', '', '', '', '', '' );
+  selectedTissue: SequencingOrder = new SequencingOrder( '', '', '', '', '', '', '', '' );
```
Here I would make properties optional or just set default values in the class constructor.

pegahtah replied:
Here I just need an empty instance to set it to what they select. I don't want it to be default values, because I want us to know if something is missing  for the Order class, and I don't want them to be optional. So I changed it to just create Empty instances. What do you think about the new changes?

pegahtah committed:
DDP-7556 PR comments

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

pegahtah committed:
DDP-7556 linting fix

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.html:
```diff
@@ -0,0 +1,36 @@
+</div>
+<ng-container *ngIf="!loading">
+  <h1> Clinical Orders </h1>
+  <div align="right" style="width: 100%; height: 100%;">
```
**#Optional**
I think it would be better, if we don't write styles inline, instead we can move them to css/scss file.

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.ts:
```diff
@@ -0,0 +1,89 @@
+        this.loading = false;
+      },
+      error: () => {
+
```
Maybe it would be better if we handle error here, or just remove this block.

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts:
```diff
@@ -545,6 +553,39 @@ export class ParticipantPageComponent implements OnInit, OnDestroy, AfterViewChe
   }
 
+  kitValueChanged( value: any, parameterName: string, sample: Sample ): void {
+    let v;
```
I would rather call variables more self-explanatory names, for example instead of "v", maybe it would be better to call it a "currentValue".

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts:
```diff
@@ -0,0 +1,162 @@
+  }
+
+  shouldHaveCheckbox( sample: SequencingOrder ): boolean {
+    return ( sample.sampleStatus === 'Received' && sample.sampleType === 'Normal' )
```
I would create an enum named SampleType and I would have stored there these string values and call them like this:
SampleType.Received, SampleType.Normal and etc.

GiCharkviani commented on ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts:
```diff
@@ -0,0 +1,162 @@
+  preSubmitOrder(): void {
+    this.errorModal = false;
+    this.modal.show();
+    return;
```
As a function return type says, this function shouldn't return anything, but instead, it returns undefined (Which is also the default return type in JS).

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

pegahtah committed:
DDP-7556 fix for spinning wheel and new role name

pegahtah committed:
DDP-7556 adding clinical orders for a participant and filters and display of them

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

pegahtah committed:
DDP-7556
*adding last status and last order number to pt page
*fix the display of PDO in pt list
*unsubscribing from subscriptions
*not showing clinical page if there is no data

pegahtah committed:
DDP-7556
*removing extra `s`

pegahtah committed:
DDP-7556
*fixing patch that got broken

pegahtah committed:
DDP-7556
*adding space for linting

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7556_Frontend

# Conflicts:
#	ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.html
